### PR TITLE
allow Symfony 7

### PR DESCRIPTION
**Pull request by dmaicher**
Closes https://github.com/Behat/Behat/issues/1441

### CODE REVIEW

**Comment by stof** [comment]
I suggest adding a condition step (running if `matrix.symfony-version == '7.0'`) doing `composer remove vimeo/psalm --no-update`.

We don't need psalm in the job running tests, and this would allow us to run the tests against Symfony 7 without waiting for Psalm to add support for Symfony 7.

**Comment by dmaicher** [comment]
Ok now comes the interesting part :smile: 

```
PHP Fatal error:  Declaration of Behat\Testwork\Cli\Application::doRun(Symfony\Component\Console\Input\InputInterface $input, Symfony\Component\Console\Output\OutputInterface $output) must be compatible with Symfony\Component\Console\Application::doRun(Symfony\Component\Console\Input\InputInterface $input, Symfony\Component\Console\Output\OutputInterface $output): int in /home/runner/work/Behat/Behat/src/Behat/Testwork/Cli/Application.php on line 98
```

**Comment by stof** [comment]
Well, Symfony 7.0 indeed forces to add return types in a bunch of places.

The good news is that variance rules added in PHP 7.2 allow us to add those return types in a child class even when using older Symfony versions (and we don't support PHP 7.1 and older).

For some classes, adding such return types might require a major version bump in Behat, but that will be clear once we know the list.

**Comment by stof** [comment]
See https://github.com/symfony/symfony/issues/51563 for the type patching utility that can add the return types automatically for many cases.

**Comment by dmaicher** [comment]
> Well, Symfony 7.0 indeed forces to add return types in a bunch of places.
> 
> The good news is that variance rules added in PHP 7.2 allow us to add those return types in a child class even when using older Symfony versions (and we don't support PHP 7.1 and older).
> 
> For some classes, adding such return types might require a major version bump in Behat, but that will be clear once we know the list.

Tests are green now :blush: From what I can tell all the classes where I added return types are final anyway? So should not be any BC break?

**Comment by stof** [comment]
@dmaicher indeed, final classes are a case where it is fine to add them.

File: composer.json [hunk 1]
```diff
@@ -18,17 +18,17 @@
...
-        "symfony/dependency-injection": "^4.4 || ^5.0 || ^6.0",
-        "symfony/event-dispatcher": "^4.4 || ^5.0 || ^6.0",
-        "symfony/translation": "^4.4 || ^5.0 || ^6.0",
-        "symfony/yaml": "^4.4 || ^5.0 || ^6.0",
+        "symfony/console": "^4.4 || ^5.0 || ^6.0 || ^7.0",
+        "symfony/config": "^4.4 || ^5.0 || ^6.0 || ^7.0",
+        "symfony/dependency-injection": "^4.4 || ^5.0 || ^6.0 || ^7.0",
+        "symfony/event-dispatcher": "^4.4 || ^5.0 || ^6.0 || ^7.0",
+        "symfony/translation": "^4.4 || ^5.0 || ^6.0 || ^7.0",
+        "symfony/yaml": "^4.4 || ^5.0 || ^6. || ^7.00",
```
[the remark this answers is not in the document]
Comment (dmaicher) [reply composer.json]: ah bad copy & paste :see_no_evil: nice catch

**Comment by alexander-schranz** [comment]
Is there something which still need to be tackled here? Maybe @dmaicher can repush as the last run was 2 month ago so we make sure it still compatible with the latest changes of Symfony 7 branch.

**Comment by dmaicher** [comment]
@alexander-schranz done :+1: 

@stof using stable dependencies now for the Symfony 7 build as well. Kept the psalm removal as v4 is incompatible with Symfony 7.

**Comment by damien-louis** [comment]
@dmaicher GG
So w're close to merge?

**Comment by loic425** [comment]
@pamil Hello Kamil, please could you merge it to help us to move forward?

**Comment by ovidiuenache** [comment]
Any update on this one?

**Comment by lchrusciel** [comment]
+1 for merging this :)

**Comment by damien-louis** [comment]
Hello. @stof ,this PR isn't mergeable?

**Review by stof — APPROVED** [review]

**Comment by lchrusciel** [comment]
SymfonyExtension is green for Sf7.0 - https://github.com/FriendsOfBehat/SymfonyExtension/pull/206

**Comment by lchrusciel** [comment]
VariadicExtension and TestExtensions works as well: https://github.com/FriendsOfBehat/VariadicExtension/pull/36

**Comment by lchrusciel** [comment]
MinkExtension is green as well: https://github.com/FriendsOfBehat/MinkExtension/pull/25

**Comment by hborras** [comment]
Is there anything else is needed to do in order to create a new release? 
I can check if something needs to be done

**Pull request merged**